src/main/java/com/strandls/user/dao/DownloadLogDao.java

```diff
@@ -0,0 +1,88 @@
+package com.strandls.user.dao;
+
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.query.Query;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.strandls.user.pojo.DownloadLog;
+import com.strandls.user.util.AbstractDAO;
+
+public class DownloadLogDao extends AbstractDAO<DownloadLog, Long> {
+
+	private static final Logger logger = LoggerFactory.getLogger(UserDao.class);
+
+	/**
+	 * @param sessionFactory
+	 */
+	@Inject
+	protected DownloadLogDao(SessionFactory sessionFactory) {
+		super(sessionFactory);
+	}
+
+	@Override
+	public DownloadLog findById(Long id) {
+		Session session = sessionFactory.openSession();
+		DownloadLog result = null;
+		try {
+			result = session.get(DownloadLog.class, id);
+		} catch (Exception e) {
+			logger.error(e.getMessage());
+		} finally {
+			session.close();
+		}
+		return result;
+	}
+
+	@SuppressWarnings({ "unchecked", "rawtypes" })
```

Comment:
why raw type is required, and why to use criteria builder, cant it be simply extracted using hql, query

Reply:
the order by field being a dynamic filed, we had to use setParameter, which is not supported for the order by, hence had to change to criteria builder to support dynamic orderby filed, Also will make the changes on rawtypes